vehicle-simulator/main.go

```diff
@@ -0,0 +1,186 @@
+var idleStateTime = flag.Int("idle-state-time", 120, "In seconds, Configure when the system will enter idle state. Idle state will send telemetry data infrequently. Receiving CTD command will reset the idle state.")
+var initialPeriodicUpdatesDelay = 0 // set the initial periodic updates delay in case of idle state active
+
+func main() {
+	flag.Parse()
+	validateFlags()
+
+	keepAlive := make(chan os.Signal, 1)
+	signal.Notify(keepAlive, os.Interrupt, syscall.SIGTERM)
+
+	initializeClient()
+	getDeviceData()
+
+	defer func() { stopTelemetrySender <- struct{}{} }()
+	defer mqttClient.Disconnect(1000)
+
+	<-keepAlive
+}
+
+func initializeClient() {
+	brokerURL := fmt.Sprintf("tcp://%s:%s", getEnvVariable("ADAPTER_BROKER_ADDRESS", "localhost"), getEnvVariable("ADAPTER_BROKER_PORT", "1883"))
+	slog.Info("Broker configuration", "URL", brokerURL)
+	options := mqtt.NewClientOptions()
+	options.AddBroker(brokerURL)
+	mqttClient = mqtt.NewClient(options)
+
+	if token := mqttClient.Connect(); token.Wait() && token.Error() != nil {
+		panic(token.Error())
+	}
+}
+
+func getEnvVariable(variableName string, defaultValue string) string {
+	value := os.Getenv(variableName)
+	if len(value) == 0 {
+		return defaultValue
+	}
+	return value
+}
+
+func getDeviceData() {
+	mqttClient.Subscribe("edge/thing/response", 1, handleDeviceDataReceived).Wait()
+	mqttClient.Publish("edge/thing/request", 1, false, "{}").Wait()
+}
+
+func handleDeviceDataReceived(sourceClient mqtt.Client, message mqtt.Message) {
+	if err := json.Unmarshal(message.Payload(), &deviceData); err != nil {
+		slog.Error("Error during payload unmarshal: ", err)
+		return
+	}
+
+	slog.Info("Connector Device Data received successfully", "data", deviceData)
+	if deviceData.DeviceID == selectedConnectorDeviceID {
+		slog.Info("Skipping the same connector deviceID instance.")
+		return
+	}
+
+	unsubscribeAllCTD()
+	subscribeForCTD(deviceData.DeviceID)
+	startSendingRandomlyGeneratedTelemetryData()
+	selectedConnectorDeviceID = deviceData.DeviceID
+}
+
+func subscribeForCTD(deviceId string) {
+	slog.Info("Subscribing for CTD commands: resetDtc & openTrunk...")
+	topicResetDtc := fmt.Sprintf("command//%s:Vehicle/req//resetDtc", deviceId)
+	ctdSubscriptionTopics = append(ctdSubscriptionTopics, topicResetDtc)
+	topicOpenTrunk := fmt.Sprintf("command//%s:Vehicle/req//openTrunk", deviceId)
+	ctdSubscriptionTopics = append(ctdSubscriptionTopics, topicOpenTrunk)
+
+	mqttClient.Subscribe(topicResetDtc, 1, handleResetDtc).Wait()
+	mqttClient.Subscribe(topicOpenTrunk, 1, handleOpenTrunk).Wait()
+}
+
+func startSendingRandomlyGeneratedTelemetryData() {
+	if !telemetrySenderStarted {
+		go telemetrySender(stopTelemetrySender, controlTelemetrySender)
+		telemetrySenderStarted = true
+	}
+}
+
+func handleResetDtc(sourceClient mqtt.Client, message mqtt.Message) {
+	if err := json.Unmarshal(message.Payload(), &deviceData); err != nil {
```

Comment:
It is strange to unmarshal Ditto message to a deviceData. If the message is to the feature, at least the path should be checked whether the command is send to the correct feature.